### PR TITLE
BUG: Series(nullable).tolist() -> numpy scalars

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

`Series.tolist` documentation states: 
> These are each a scalar type, which is a Python scalar (for str, int, float) or a pandas scalar (for Timestamp/Timedelta/Interval/Period)

Nullable dtype behavior seems to be inconsistent with the docstring and other dtypes:

main:
```
import pandas as pd

type(pd.Series([1], dtype="Int64").tolist()[0])             # numpy.int64
type(pd.Series([1], dtype="int64").tolist()[0])             # int
type(pd.Series([1], dtype="int64[pyarrow]").tolist()[0])    # int
```

PR:
```
import pandas as pd

type(pd.Series([1], dtype="Int64").tolist()[0])             # int
type(pd.Series([1], dtype="int64").tolist()[0])             # int
type(pd.Series([1], dtype="int64[pyarrow]").tolist()[0])    # int
```

The fix also provides an improvement in performance:

```
import pandas as pd
import numpy as np

ser = pd.Series(np.arange(10**6), dtype="Int64")

%timeit ser.tolist()

# 45.5 ms ± 980 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  <- main
# 28.5 ms ± 229 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  <- PR
```

### CODE REVIEW

**Review by phofl — COMMENTED** [review]
Could you profile this for pyarrow as well and with NA?

**Comment by jreback** [comment]
hmm i seem to recall that we very specifically have python scalers in all of these list conversion methods (eg iter as well)

i would be surprised this doesn't break many things

**Comment by lukemanley** [comment]
| Could you profile this for pyarrow as well and with NA?

pyarrow doesn't hit this method. It hits ExtensionArray.tolist. I have a separate PR in the works to make that faster.

Here is this PR profiled with NA:

```
import pandas as pd
import numpy as np

ser = pd.Series(np.arange(10**6), dtype="Int64")
ser.iloc[::2] = pd.NA

%timeit ser.tolist()

# 92.9 ms ± 1.72 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)  <- main
# 59.5 ms ± 4.72 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)  <- PR
```

**Comment by lukemanley** [comment]
> edit: i think i misread your post
you are making things consistent ok!

Yes, sorry my wording should have been more clear. This makes nullable return python scalars which is consistent with other dtypes

**Comment by phofl** [comment]
Thx, missed the file name. good that casting to object is still faster

**Comment by phofl** [comment]
Could you merge main? We got some deprecation warnings from numpy that are fixed on main

File: pandas/tests/series/methods/test_tolist.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        (["abc"], "string", str),
+        ([Interval(1, 3)], "interval", Interval),
+        ([Period("2000-01-01", "D")], "period[D]", Period),
+        ([Timedelta(days=1)], "timedelta64[ns]", Timedelta),
+        ([Timestamp("2000-01-01")], "datetime64[ns]", Timestamp),
+        pytest.param(
+            [1],
+            "int64[pyarrow]",
+            int,
+            marks=td.skip_if_no("pyarrow", min_version="1.0.0"),
```
Comment (phofl) [inline pandas/tests/series/methods/test_tolist.py:31]: I think min version is 7 or something similar, please adjust accordingly

Comment (lukemanley) [reply pandas/tests/series/methods/test_tolist.py]: updated to version 6 which is the min version tested, although this test should pass with any version. It looks like there are a handful of other tests that have less than 6.0.

Comment (phofl) [reply pandas/tests/series/methods/test_tolist.py]: I think you can get rid of the min version then

Comment (lukemanley) [reply pandas/tests/series/methods/test_tolist.py]: ah, thanks. updated

**Review by mroeschke — APPROVED** [review]
LGTM merge when ready @phofl

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @lukemanley

**Pull request merged**